src/dag/write.ts

```diff
@@ -3,39 +3,43 @@ import {chunkDataKey, chunkMetaKey, headKey, chunkRefCountKey} from './key';
 import {Read} from './read';
 import {assertMeta, Chunk, ChunkHasher} from './chunk';
 import {assertNumber} from '../asserts';
-import {assertNotTempHash, Hash} from '../hash';
+import type {Hash} from '../hash';
 
 type HeadChange = {
   new: Hash | undefined;
   old: Hash | undefined;
 };
 
 export class Write extends Read {
-  private readonly _kvw: kv.Write;
+  protected declare readonly _kvr: kv.Write;
```

Comment:
I did not know you could use declare to do a covariant override of a super classes property type.

I think this is more confusing than the previous approach of just storing the write in the private `_kvw` property.   Calling write methods on a property named `_kvr` (abbreviation for kv.Read) is odd.

Reply:
I though I was being smart and not adding another property but maybe it is just too silly?